StyleCop.Analyzers/StyleCop.Analyzers.Test/LayoutRules/SA1509UnitTests.cs

```diff
@@ -0,0 +1,702 @@
+        [TestMethod]
+        public async Task TestEmptySource()
+        {
+            var testCode = string.Empty;
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationOpeningBraceHasBlankLine()
+        {
+            var testCode = @"
+class Foo
+
+{
+
+}";
+
+            var expected = new[]
+                {
+                    new DiagnosticResult
+                    {
+                        Id = this.DiagnosticId,
+                        Message = "Opening curly brackets must not be preceded by blank line.",
+                        Severity = DiagnosticSeverity.Warning,
+                        Locations =
+                            new[]
+                            {
+                                new DiagnosticResultLocation("Test0.cs", 4, 1)
+                            }
+                    }
+                };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationOpeningBraceDoesntHaveBlankLine()
+        {
+            var testCode = @"
+class Foo
+{
+
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationCommentBeforeOpeningBrace()
+        {
+            var testCode = @"
+class Foo
+//this is a comment
+{
+
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationMultilineCommentBeforeOpeningBrace()
+        {
+            var testCode = @"
+class Foo
+/*this is a comment
+that spans 2 lines */
+{
+
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [TestMethod]
+        public async Task TestClassDeclarationMultilineCommentBeforeOpeningBraceButBlankLineExists()
+        {
+            var testCode = @"
+class Foo
+/*this is a comment
+that spans 2 lines */
+
+//another comment
+{
+
+}";
+
+            var expected = new[]
+                {
+                    new DiagnosticResult
+                    {
+                        Id = this.DiagnosticId,
+                        Message = "Opening curly brackets must not be preceded by blank line.",
+                        Severity = DiagnosticSeverity.Warning,
+                        Locations =
+                            new[]
+                            {
+                                new DiagnosticResultLocation("Test0.cs", 7, 1)
+                            }
+                    }
+                };
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, expected, CancellationToken.None);
+        }
+
```

Comment:
When we do notice behavior deviations, we try to get them at least added to #359. That said, I don't think we should report this case as a diagnostic right now for the following reasons:

1. Users are likely to see this as *not* being preceded by a blank line
2. The code fix for this might not have the desired behavior

It's probably something we can revisit in the future if you find a related case where you do want to see a warning reported.